java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,602 @@
+          }
+          currentBlock = createUnconditionalJump(tree, breakTargets.getLast());
+        } else {
+          currentBlock = createUnconditionalJump(tree, null);
+          gotos.add(currentBlock);
+        }
+        break;
+      }
+      case CONTINUE_STATEMENT: {
+        ContinueStatementTree continueStatementTree = (ContinueStatementTree) tree;
+        if(continueStatementTree.label() == null) {
+          if (continueTargets.isEmpty()) {
+            throw new IllegalStateException("'break' statement not in loop or switch statement");
+          }
+          currentBlock = createUnconditionalJump(tree, continueTargets.getLast());
+        } else {
+          currentBlock = createUnconditionalJump(tree, null);
+          gotos.add(currentBlock);
+        }
+        break;
+      }
+      case WHILE_STATEMENT: {
+        WhileStatementTree s = (WhileStatementTree) tree;
+        Block falseBranch = currentBlock;
+        Block loopback = createBlock();
+        // process body
+        currentBlock = createBlock(loopback);
+        continueTargets.addLast(loopback);
+        breakTargets.addLast(falseBranch);
+        build(s.statement());
+        breakTargets.removeLast();
+        continueTargets.removeLast();
+        Block bodyBlock = currentBlock;
+        // process condition
+        currentBlock = createBranch(s, bodyBlock, falseBranch);
+        buildCondition(s.condition(), bodyBlock, falseBranch);
+        loopback.successors.add(currentBlock);
+        currentBlock = createBlock(currentBlock);
+        break;
+      }
+      case DO_STATEMENT: {
+        DoWhileStatementTree s = (DoWhileStatementTree) tree;
+        Block falseBranch = currentBlock;
+        Block loopback = createBlock();
+        // process condition
+        currentBlock = createBranch(s, loopback, falseBranch);
+        buildCondition(s.condition(), loopback, falseBranch);
+        // process body
+        currentBlock = createBlock(currentBlock);
+        continueTargets.addLast(loopback);
+        breakTargets.addLast(falseBranch);
+        build(s.statement());
+        breakTargets.removeLast();
+        continueTargets.removeLast();
+        loopback.successors.add(currentBlock);
+        currentBlock = createBlock(currentBlock);
+        break;
+      }
+      case FOR_EACH_STATEMENT: {
+        ForEachStatement s = (ForEachStatement) tree;
+        //TODO(npe) One solution is to create a forstatement node depending on type of expression (iterable or array) and build CFG from it.
+        Block afterLoop = currentBlock;
+        Block loopback = createBlock();
+        continueTargets.addLast(loopback);
+        breakTargets.addLast(afterLoop);
+        build(s.statement());
+        breakTargets.removeLast();
+        continueTargets.removeLast();
+        build(s.variable());
+        build(s.expression());
+        break;
+      }
+      case FOR_STATEMENT: {
+        ForStatementTree s = (ForStatementTree) tree;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 11. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)